### ISSUE TITLE
Prepare rendering (phase #1)

### ISSUE THREAD

**Issue by ruslangaripov** [Serg-Norseman/Arbor#3]
The library must:
- Exports C++ class or function (the first is more preferable) that wraps a HWND created by a dedicated GUI thread. The HWND is a place where the graph is rendered. Client of the HWND controls the window styles (general and extended). Client of the HWND controls the window with Windows API calls. Client ain't aware of the GUI thread owning the HWND; the library controls the thread by itself.
- ~~The class or another function must give ability to set initial data or add new data to the graph~~ (too early to implement in this ticket).

**Comment by ruslangaripov** [comment]
:runner:

**Comment by ruslangaripov** [comment]
:memo:  Some notes about current implementation (_it ain't yet pushed into any publicly available branch_):
- Targeting to Windows 7 I use Direct2D 1.1 only when render on HWND surfaces (that's true for the library and sample application).
- DXGI swap chain doesn't use VSync. The code relies on DWM -- it must do that. Actually rendering in the sample application may use DXGI with VSync turned on 'cos the sample has no animation. And I'm planning to change it in the sample. But I believe the DLL is gonna use animation. Therefore currently I turn VSync off in the DLL code. That's why I also don't use "scrolling optimization" for DXGI swap chain presentation.
- DLL's window (where the graph will be rendered) initializes WAM though the window doesn't use it currently.

**Comment by ruslangaripov** [comment]
:memo: Method(s) for setting/adding data will be part(s) of other ticket(s). It's too early now to implement it.

**Comment by ruslangaripov** [comment]
:memo: Notes for [6b6d19e](https://github.com/ruslangaripov/Arbor/commit/6b6d19ef6a1b75a75c4fb1a88c0e013441f1c78d) commit:
Fix ticket #3 (add sample rendering, scrolling etc). High DPI tests passed OK.
Will close the ticket later, after some additional tests.

P.S. Never use merge-commit command in git :astonished: you lose commits from merged branch :stuck_out_tongue_closed_eyes:

**Comment by ruslangaripov** [comment]
:memo: Notes for [f6cab78](https://github.com/ruslangaripov/Arbor/commit/f6cab7818eafb95e7c7486a90a1a50febf7aaa5e) commit:
- Both tickets #3 and #4 fixed,
- The graph window (provided by the DLL) renders _a sample_ -- mesh and ellipse. This will be removed in the subsequent commits,
- The sample application provides optional scroll bars (see ticket #4). It ain't a model implementation but a simple one,
- The code passed tests for High DPI and per-monitor DPI settings,
- Been compiled by MSVC 2015 Update 1 the code has some noisy warnings from `MSVC prefast'. In particular:

```
... cpp\source\arborgvt\service\winapi\srwlock.h(24): warning C26135: Missing annotation _Acquires_exclusive_lock_(this->m_lock) at function 'wapi::srw_lock::lockExclusive'.
... cpp\source\arborgvt\service\winapi\srwlock.h(36): warning C26135: Missing annotation _Releases_exclusive_lock_(this->m_lock) at function 'wapi::srw_lock::unlockExclusive'.
... cpp\source\arborgvt\service\winapi\srwlock.h(65): warning C26135: Missing annotation _Acquires_exclusive_lock_(this->m_lock) at function 'wapi::srw_lock::tryLockExclusive'.
... cpp\source\arborgvt\service\winapi\srwlock.h(71): warning C26135: Missing annotation _Acquires_shared_lock_(this->m_lock) at function 'wapi::srw_lock::tryLockShared'.
```

Visual Studio drags those warnings starting from Visual Studio 2013. It still ain't fixed. :open_mouth: See comments inside `srwlock.h' file for more details about it. (Both project have enabled prefast for 'Release' configuration)
